File: benchmark/Lua 5.3/百万素数查找耗时.lua
local total = 0
function is_prime(n) 
	local count = math.sqrt(n)+1
    for x = 2, count do
        if (n % x) == 0 then 
			return false 
		end
    end
	total = total + 1
    return true
end

local start = os.clock()

for i = 0, 1000000-1, 1 do
	is_prime(i)
end
print(total, "\n");

print(os.clock()-start, "\n")

File: benchmark/CZL 1.0/百万素数查找耗时.c
var total = 0
int isprime(int n)
{
	count = int(os.sqrt(n))
	for i in (2, count)
		if (n%i == 0)
			return false
	$total++
	return true
}

main
{
	for i in (0, 1000000-1)
		isprime(i)
	
	echo($total)
	echo(clock())
}
